gtfs/spec/en/reference.md

```diff
@@ -29,174 +29,176 @@ This document explains the types of files that comprise a GTFS transit feed and
 
-* **Field required** - The field column must be included in your feed, and a value must be provided for each record. Some required fields permit an empty string as a value. To enter an empty string, just omit any text between the commas for that field. Note that 0 is interpreted as "a string of value 0", and is not an empty string. Please see the field definition for details.
-* **Field optional** - The field column may be omitted from your feed. If you choose to include an optional column, each record in your feed must have a value for that column. You may include an empty string as a value for records that do not have values for the column. Some optional fields permit an empty string as a value. To enter an empty string, just omit any text between the commas for that field. Note that 0 is interpreted as "a string of value 0", and is not an empty string.
-* **Field conditionally required** - This field or file is **required** under certain conditions, which are outlined in the field or file *Details*. Outside of these conditions, this field or file is optional. 
-* **Dataset unique** - The field contains a value that maps to a single distinct entity within the column. For example, if a route is assigned the ID **1A**, then no other route may use that route ID. However, you may assign the ID **1A** to a location because locations are a different type of entity than routes.
+* **Feed** - Successive datasets comprise a feed.
+* **Dataset** - A complete set of files defined by this specification reference. Datasets should be published at a public, permanent URL, including the zip file name. (e.g., www.agency.org/gtfs/gtfs.zip). For more information see GTFS best practices. 
+* **Field** - A specific type of information for records in a .txt file. Represented, in a table, as a column.
+* **Record** - A basic data structure comprised of a number of different fields describing a single entity (e.g. transit agency, stop, route, etc.). Represented, in a table, as a row. 
+* **Field Value** - An individual entry in a field. Represented, in a table, as a single cell. 
+* **Required** - The field must be included in the dataset, and a value must be provided for each field. Some required fields permit an empty string as a value (denoted in this specification as "empty"). To enter an empty string, just omit any text between the commas for that field. Note that `0` is interpreted as "a string of value 0", and is not an empty string. See the field definition for details.
+* **Optional** - The field may be omitted from the dataset. If an optional column is included, each field in the dataset must have a value for that column. An empty string may be included as a value for fields that do not have values for the column. Some optional fields permit an empty string as a value. To enter an empty string, just omit any text between the commas for that field. Note that `0` is interpreted as "a string of value 0", and is not an empty string.
```

Comment:
If we're eliminating the phrase 
>"Note that `0` is interpreted as "a string of value 0", and is not an empty string."
in the *Required* definition, shouldn't we also eliminate it here?

Reply:
> each field in the dataset must have a value for that column.

Wait. What. I don't get it. No, it's optional.

Reply:
We should not consider empty as being a value. empty is a lack of value. Otherwise sentences are bitting their own tails. 

No value = empty value.
Required value = non-empty value.

Reply:
>each field in the dataset must have a value for that column.

I'm not sure about their either.  I agree that the value should be optional (not required), but IIRC there are some places in the GTFS spec where this doesn't currently hold true (i.e., optional fields, if provided, must have values).

@giocorti If the above is true, could you list some examples?

Reply:
@barbeau In the current version of the spec `arrival_time` and `departure_time` both say
>If interpolated times are indicated with timepoint=0, then time points must be indicated with a value of 1 for the timepoint field.

I'll let you know if I find more examples.

Reply:
> We should not consider empty as being a value. empty is a lack of value. Otherwise sentences are bitting their own tails.
> 
> No value = empty value.
> Required value = non-empty value.

If we do this I think it'll create a few issues that we'll need to settle:
1. We'll need clear language to differentiate between a field value having no value because it is empty and a field value having no value because the field is omitted. 
2. On a related note we'll need to address whether GTFS-consuming applications should ever make different interpretations about omitted vs. empty field values. 
3. We'll need to rewrite a few of the definitions such as the "required" definition. I'm currently working on this and I'll float a few ideas soon. 

@LeoFrachet, @barbeau thoughts? Especially on the first two?